src/client/content_mapblock.cpp

```diff
@@ -404,9 +384,99 @@ void MapblockMeshGenerator::drawAutoLightedCuboid(aabb3f box, const f32 *txc,
 			d.Z = (j & 1) ? dz2 : dz1;
 			lights[j] = blendLight(d);
 		}
-		drawCuboid(box, tiles, tile_count, lights, txc, mask);
+
+		drawCuboid(box, tiles, tile_count, txc, mask, [&] (int face, video::S3DVertex vertices[4]) {
+			for (int j = 0; j < 4; j++) {
+				video::S3DVertex &vertex = vertices[j];
+				vertex.Color = encode_light(
+					lights[light_indices[face][j]].getPair(MYMAX(0.0f, vertex.Normal.Y)),
+					f->light_source);
+				if (!f->light_source)
+					applyFacesShading(vertex.Color, vertex.Normal);
+			}
+		});
+	} else {
+		drawCuboid(box, tiles, tile_count, txc, mask, [&] (int face, video::S3DVertex vertices[4]) {
+			video::SColor color = encode_light(light, f->light_source);
+			if (!f->light_source)
+				applyFacesShading(color, vertices[0].Normal);
+			for (int j = 0; j < 4; j++) {
+				video::S3DVertex &vertex = vertices[j];
+				vertex.Color = color;
+			}
+		});
+	}
+}
+
+void MapblockMeshGenerator::drawSolidNode()
+{
+	int faces = 0; // k-th bit will be set if k-th face is to be drawn.
```

Comment:
`u8`

Reply:
Updated. That saved 8 bytes of object code!

Reply:
Note: at -O3 my gcc optimizes it to `int` +)